corgi/core/models.py

```diff
@@ -347,70 +347,39 @@ def get_related_components(build_ids: QuerySet, sort_field: str) -> QuerySet["Co
         # Else self.builds is an empty QuerySet
         return Component.objects.none()
 
-    @staticmethod
-    def get_product_component_relations(
-        variant_ids: list[str], sort_field="external_system_id", only_errata=False
-    ) -> QuerySet["ProductComponentRelation"]:
-        """Returns unique productcomponentrelations with at least 1 matching variant_id,
-        ordered by sort_field. if only_errata is True, only return PCRs with Type=ERRATA
-        """
-        if variant_ids:
-            query = Q()
-            for variant_id in variant_ids:
-                query |= Q(product_ref__contains=variant_id)
-            result = ProductComponentRelation.objects.filter(query).distinct().order_by(sort_field)
-            if only_errata:
-                result = result.filter(type=ProductComponentRelation.Type.ERRATA)
-            return result
-        # Else self.product_variants is an empty QuerySet
-        return ProductComponentRelation.objects.none()
-
     @property
     def builds(self) -> QuerySet:
-        """Return unique build IDs for errata with variant_ids matching self.product_variants"""
-        return (
-            self.get_product_component_relations(
-                self.product_variants, "build_id", only_errata=True
+        """Returns unique productcomponentrelations with at least 1 matching variant or stream,
+        ordered by build_id.
+        """
+        product_refs = []
+        if isinstance(self, ProductVariant):
+            product_refs.append(self.name)
+        elif isinstance(self, ProductStream):
+            product_refs.append(self.name)
+            # we also want to include child product_variants for each product stream in the query
+            product_refs.extend(self.product_variants)
```

Comment:
I'm not clear why we're using self.name for a ProductStream, but not for a Product or a ProductVersion.

Assuming the product_streams / product_variants property is defined as an empty list on the ProductStream / ProductVariant model:
```suggestion
        # for all ProductModel subclasses, use the list of child streams and child variants
        product_refs = self.product_streams
        product_refs += self.product_variants
        # for streams, "child streams" is an empty list so add the stream's name
        # for variants, "child variants" is an empty list so add the variant's name
        if isinstance(self, ProductStream) or isinstance(self, ProductVariant):
            product_refs += [self.name]
```

Or maybe:
```suggestion
        # for all ProductModel subclasses, look up based on current name
        product_refs = [self.name]
        # for ProductStream and ProductVariant classes, one of below will be an empty list
        product_refs += self.product_streams
        product_refs += self.product_variants
```

Reply:
Great feedback, thank you. I'll go with the one-liner here.

Reply:
On section thought and closer inspection we don't want to include product_streams list when the type is ProductVariant, because that is too general. I'm leaving this closer to the original code with a small tweak to avoid concatenating None and list.

Reply:
I added some tests for Product and ProductVersion nested builds.